### PR TITLE
Bytecode file corner case fixes in "Native" JS executor.

### PR DESCRIPTION
**Pull request by reseul**
1. Sometimes an application closes too fast for the "bytecode file generation" task to complete. The result is  usually a smaller (incomplete) file that triggers an access denied exception next run when chakra tries to load it.
Switched to a "write to temp file and rename" solution.

2. Made sure any SEH exception during reading of bytecode files deletes those files as well.
"Soft" errors returned by the JS API itself continue to be dealt with on the managed side (where the bytecode file is also deleted)

3. Fixed some memory leaks

### CODE REVIEW

File: ReactWindows/ChakraBridge/ChakraHost.cpp [hunk 1]
```diff
@@ -333,21 +334,50 @@ JsErrorCode ChakraHost::SerializeScript(const wchar_t* szPath, const wchar_t* sz
...
+
+        size_t tmpFilePathSize = wcslen(szSerializedPath) + 4; // add string size of ".tmp"
+        szTmpSerializedPath = new wchar_t[tmpFilePathSize + 1];
+        swprintf(szTmpSerializedPath, tmpFilePathSize + 1, L"%ws.tmp", szSerializedPath);
 
         FILE* file;
-        _wfopen_s(&file, szSerializedPath, L"wb");
-        fwrite(buffer, sizeof(BYTE), bufferSize, file);
+        errno_t err;
+        err = _wfopen_s(&file, szTmpSerializedPath, L"wb");
```
Comment (matthargett) [inline ReactWindows/ChakraBridge/ChakraHost.cpp]: combine declaration and assignment. also, any reason to avoid use of `auto`?

Comment (reseul) [reply ReactWindows/ChakraBridge/ChakraHost.cpp]: Will change. No reason, perhaps I'm just not very accustomed to it (I'm an old timer)

**Comment by rozele** [comment]
@matthargett - we're missing a bunch of documentation on how to change the JS executor on react-native-windows. We need a separate PR to include that documentation so there would even be a place to make that change.

**Comment by reseul** [comment]
Fixed the "auto" and added a comment in the code.

**Comment by reseul** [comment]
@rozele I'm putting this on hold, I'm investigating some issues...

**Comment by reseul** [comment]
@rozele I think this can go in.

**Pull request merged**